chore: rename snapshot proto package

julienrbrt opened the pull request:
<!--
The default pull request template is for types feat, fix, or refactor.
For other templates, add one of the following parameters to the url:
- template=docs.md
- template=other.md
-->

## Description

See https://github.com/cosmos/cosmos-sdk/pull/14591#discussion_r1068044503
As per https://github.com/cosmos/cosmos-sdk/pull/14597#issuecomment-1380678894 this PR renames the package to `v1` but not re-add the fields compared to https://github.com/cosmos/cosmos-sdk/pull/14608 (on release/v0.47.x)

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] targeted the correct branch (see [PR Targeting](https://github.com/cosmos/cosmos-sdk/blob/main/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] followed the guidelines for [building modules](https://github.com/cosmos/cosmos-sdk/blob/main/docs/docs/building-modules)
- [ ] included the necessary unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/main/CONTRIBUTING.md#testing)
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed `!` in the type prefix if API or client breaking change
- [ ] confirmed all author checklist items have been addressed 
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

julienrbrt commented on proto/cosmos/base/store/snapshots/v1beta1/snapshot.proto:
```diff
@@ -1,9 +1,9 @@
 syntax = "proto3";
-package cosmos.base.store.snapshots.v1beta1;
+package cosmos.base.snapshots.v1beta1;
```
Wrong copy/paste, will fix.

julienrbrt commented:
One thing actually, we've changed the package name in 0.48 vs <= 0.47.
So it will be proto breaking anyway in v0.48 I think.

amaurym commented:
> One thing actually, we've changed the package name in 0.48 vs <= 0.47 (https://github.com/cosmos/cosmos-sdk/commit/3222be3c84bbefda00255d49bd2d93d4acba87be).
So it will be proto breaking anyway in v0.48 I think.

Oh, if we changed package names, then we can remove fields freely in the newly-named package. It's not actually a rename, but basically leaving the old one (`cosmos.base.snapshots.v1beta1`) as-is (it's on buf anyways), and creating a new one `cosmos.base.store.snapshots.v1beta1`. So we're free to do anything in the new one.

Also, maybe we can use `v1` for the new one?

julienrbrt commented:
> > One thing actually, we've changed the package name in 0.48 vs <= 0.47 ([3222be3](https://github.com/cosmos/cosmos-sdk/commit/3222be3c84bbefda00255d49bd2d93d4acba87be)).
> > So it will be proto breaking anyway in v0.48 I think.
> 
> Oh, if we changed package names, then we can remove fields freely in the newly-named package. It's not actually a rename, but basically leaving the old one `cosmos.base.snapshots.v1beta1` as-is (it's on buf anyways), and creating a new one `cosmos.base.store.snapshots.v1beta1`. So we're free to do anything in the new one.
> 
> Also, maybe we can use `v1` for the new one?

Okay, so I am not re-adding it here, but only rename it to v1 and in v0.47 (where it is not yet renamed, I can add it as deprecated)

julienrbrt commented on CHANGELOG.md:
```diff
@@ -154,6 +154,7 @@ Ref: https://keepachangelog.com/en/1.0.0/
 
 ### API Breaking Changes
 
+* (snapshots) [#14597](https://github.com/cosmos/cosmos-sdk/pull/14597) Move `snapshots` to `store/snapshots` and rename bump proto package to v1.
```
We had no changelog for the package moving (I am noticing that we mostly forgot to a changelog all store refactors). Precising the renaming as well.

tac0turtle reviewed (commented):
do we want to keep the old one around?

julienrbrt commented:
> do we want to keep the old one around?

Don't think it's necessary as it isn't used.

amaurym commented on proto/cosmos/base/store/snapshots/v1/snapshot.proto:
```diff
@@ -1,5 +1,5 @@
 syntax = "proto3";
-package cosmos.base.store.snapshots.v1beta1;
+package cosmos.base.store.snapshots.v1;
```
```suggestion
package cosmos.store.snapshots.v1;
```

Small proposal for a shorter name. 👍 / 👎 ?

julienrbrt replied:
cc @tac0turtle

tac0turtle replied:
it looks nice but it fails proto lint since the linter wants the path to the be the package

amaurym replied:
Right, we would just need to move the *.proto files to a folder higher.

tac0turtle replied:
yea open to that

julienrbrt replied:
Done in https://github.com/cosmos/cosmos-sdk/pull/14597/commits/9d3b6204ce4569a2b0d689fcd23fa7b949d6002d
